src/Sylius/Component/Taxonomy/Factory/TaxonomyFactory.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Component\Taxonomy\Factory;
+
+use Sylius\Component\Resource\Factory\FactoryInterface;
+use Sylius\Component\Core\Model\Taxonomy;
+use Sylius\Component\Taxonomy\Model\TaxonomyInterface;
+use Sylius\Component\Translation\Factory\TranslatableFactory;
+use Sylius\Component\Translation\Provider\LocaleProviderInterface;
+
+/**
+ * @author Magdalena Banasiak <magdalena.banasiak@lakion.com>
+ */
+class TaxonomyFactory extends TranslatableFactory implements FactoryInterface
+{
+    /**
+     * @var FactoryInterface
+     */
+    private $taxonFactory;
+
+    /**
+     * @param FactoryInterface $taxonFactory
+     */
+    public function __construct(FactoryInterface $factory, LocaleProviderInterface $localeProvider, FactoryInterface $taxonFactory)
```

Comment:
What is `$factory` here? Why do we even need the first two arguments? /cc @pjedrzejewski